github/Requester.py

```diff
@@ -999,19 +1061,24 @@ def __requestRaw(
         url: str,
         requestHeaders: Dict[str, str],
         input: Optional[Any],
-    ) -> Tuple[int, Dict[str, Any], str]:
+        stream: bool = False,
+    ) -> Tuple[int, Dict[str, Any], Union[str, requests.Response]]:
         self.__deferRequest(verb)
 
         try:
             original_cnx = cnx
             if cnx is None:
                 cnx = self.__createConnection()
-            cnx.request(verb, url, input, requestHeaders)
-            response = cnx.getresponse()
-
-            status = response.status
-            responseHeaders = {k.lower(): v for k, v in response.getheaders()}
-            output = response.read()
+            if stream:
+                response = cnx.session.get(url, stream=True, headers=requestHeaders)
+                status = response.status_code
+                responseHeaders = {k.lower(): v for k, v in response.headers.items()}
+            else:
+                cnx.request(verb, url, input, requestHeaders)
+                resp = cnx.getresponse()
+                output = resp.read()
+                status = resp.status
+                responseHeaders = {k.lower(): v for k, v in resp.getheaders()}
```

Comment:
I think we can simplify this:
```suggestion
            response = cnx.request(verb, url, input, requestHeaders, stream=stream)
            resp = cnx.getresponse()
            output = resp.iter_content() if stream else output = resp.read()
            status = resp.status_code
            responseHeaders = {k.lower(): v for k, v in resp.getheaders()}
```

This requires some modification to `HTTPRequestsConnectionClass` / `HTTPSRequestsConnectionClass` in https://github.com/PyGithub/PyGithub/blob/main/github/Requester.py.

Reply:
Okay, I think I see what you are saying.  It would likely be more consistent between stream and not stream, so maybe easier to understand.